subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -508,6 +508,10 @@ static int topic_subscribe(void)
 		}
 	}
 
+	struct aws_iot_evt aws_iot_evt = { 0 };
```

Comment:
At this point, we don't know if a successful subscription to the topics has been made or not because we don't check the err variable returned by the ```mqtt_subscribe``` functions. I think a more suitable place for the ```AWS_IOT_EVT_READY``` notification would be after the call to ```topics_subscribe()``` in the ```MQTT_CONNACK```. The notification of the ```AWS_IOT_EVT_CONNECTED``` event can be moved to before ```topics_subscribe()``` to keep the order in which the events are omitted today. We should also add an error check of the ```topics_subscribe``` function, making it return if it fails. Something like this:
```
		LOG_DBG("MQTT client connected!");

		aws_iot_evt.data.persistent_session =
				   mqtt_evt->param.connack.session_present_flag;
		aws_iot_evt.type = AWS_IOT_EVT_CONNECTED;
		aws_iot_notify_event(&aws_iot_evt);

		if (!mqtt_evt->param.connack.session_present_flag) {
			err = topic_subscribe();
			if (err) {
				aws_iot_evt.type = AWS_IOT_EVT_ERROR;
				aws_iot_evt.data.err = err;
				aws_iot_notify_event(&aws_iot_evt);
				break;
			}
		}

		aws_iot_evt.type = AWS_IOT_EVT_READY;
		aws_iot_notify_event(&aws_iot_evt);
		break;
```

Reply:
Ok, I have changed it.